plugins/modules/ndo_interface_setting.py

```diff
@@ -0,0 +1,864 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2024, Anvitha Jain (@anvjain) <anvjain@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: ndo_interface_setting
+short_description: Manage Interface Policy Groups on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage Interface Policy Groups on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Anvitha Jain (@anvjain)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be a fabric policy template.
+    type: str
+    required: true
+  interface_policy_group:
+    description:
+    - The name of the interface policy group.
+    type: str
+    aliases: [ name ]
+  interface_policy_group_uuid:
+    description:
+    - The UUID of the interface policy group.
+    - This parameter is required when the O(interface_policy_group) needs to be updated.
+    type: str
+    aliases: [ uuid ]
+  description:
+    description:
+    - The description of the interface policy group.
+    type: str
+  interface_type:
+    description:
+    - The type of the interface policy group.
+    - The default value is C(physical).
```

Comment:
When `interface_type` defaults to `physical`? Please add a task to create the "Interface Policy Group" object without using `interface_type`.

Reply:
interface_type defaults to physical, removed this as it is not supported by the API